scrapy/crawler.py

```diff
@@ -71,7 +71,6 @@ def __init__(
 
         self.spidercls: Type[Spider] = spidercls
         self.settings: Settings = settings.copy()
-        self.spidercls.update_settings(self.settings)
 
         self.addons: AddonManager = AddonManager(self)
         self.addons.load_settings(self.settings)
```

Comment:
I am also thinking that addons could be a way to cleaning implement argument processing this way.

If we move addons after spider.update_settings, addons can use crawler.spider from from_crawler to process spider args. It would happen after spider.__init__, but other than that, it would be cleaner than needing code in a base spider, a spider mixin, or a spider itself.